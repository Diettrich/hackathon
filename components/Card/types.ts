export type PlayerProps = {
  firstname: string;
  lastname: string;
  goal: number;
  salary: number;
  devise: string;
  pictureURl: string;
};
