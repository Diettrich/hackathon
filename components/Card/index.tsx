import React from "react";
import { motion } from "framer-motion";
import Image from "next/image";

import { PlayerProps } from "./types";

import styles from "./styles.module.scss";

const Card: React.FC<PlayerProps> = ({
  firstname,
  lastname,
  goal,
  salary,
  devise,
  pictureURl,
}) => {
  const containerVariants = {
    hidden: { opacity: 0 },
    visible: { opacity: 1, transition: { duration: 1 } },
  };

  return (
    <motion.div
      className={styles.container}
      variants={containerVariants}
      initial="hidden"
      animate="visible"
      data-testid="card-container"
    >
      <div className={styles.pictureContainer}>
        <Image
          data-testid="card-picture"
          className={styles.picture}
          src={pictureURl}
          alt={`${firstname} ${lastname}`}
          height={222}
          width={300}
        />
      </div>
      <div className={styles.nameContainer}>
        <h2
          className={styles.name}
          data-testid="card-name"
        >{`${firstname} ${lastname}`}</h2>
      </div>
      <div className={styles.detailsContainer}>
        <div className={styles.goal} data-testid="card-goal">
          <span className={styles.label}>Goal: </span>
          <span className={styles.value}>{`${goal} ${devise}`}</span>
        </div>
        <div className={styles.salary} data-testid="card-salary">
          <span className={styles.label}>Salary: </span>
          <span className={styles.value}>{`${salary} ${devise}`}</span>
        </div>
      </div>
    </motion.div>
  );
};

export default Card;
