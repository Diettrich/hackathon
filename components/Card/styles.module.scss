.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 300px;
  height: 300px;
  background-color: #f5f5f5;
  border-radius: 1rem;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  padding-bottom: 1rem;
  text-align: center;

  @media (max-width: 640px) {
    width: 100%;
    height: auto;
    padding: 0;
  }
}

.pictureContainer {
  width: 100%;
  height: auto;
  max-height: 222px;
  border-radius: 1rem 1rem 0 0;
  overflow: hidden;
  margin-bottom: 1rem;
}

.picture {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.nameContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.name {
  margin: 0;
  font-size: 1.5rem;
  font-weight: bold;
  text-align: center;
  margin-left: 1rem;
  color: darkblue;
}

.detailsContainer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-top: 1rem;

  @media (max-width: 640px) {
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin-top: 1rem;
  }
}

.goal,
.salary {
  display: flex;
  align-items: center;
  margin: 0.5rem 0;
  text-align: center;
  width: 100%;
  font-weight: 500;
}

.label {
  font-weight: bold;
  margin-right: 0.5rem;
  color: gray;
}

.value {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
