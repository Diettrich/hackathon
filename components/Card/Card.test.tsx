import React from "react";
import { render, screen } from "@testing-library/react";
import Card from "./index";

describe("Card component", () => {
  const props = {
    firstname: "John",
    lastname: "Doe",
    goal: 5000,
    salary: 2500,
    devise: "$",
    pictureURl: "https://example.com/image.jpg",
  };

  test("renders the component with correct data", () => {
    render(<Card {...props} />);

    const name = screen.getByTestId("card-name");
    const goal = screen.getByTestId("card-goal");
    const salary = screen.getByTestId("card-salary");
    const picture = screen.getByTestId("card-picture");

    expect(name).toHaveTextContent("John Doe");
    expect(goal).toHaveTextContent("Goal: 5000 $");
    expect(salary).toHaveTextContent("Salary: 2500 $");
    expect(picture).toHaveAttribute(
      "src",
      "/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
    ); // since its a Next Image component, Later we can add a sanitizer to get it ready from a function
    expect(picture).toHaveAttribute("alt", "John Doe");
  });

  test("renders the component with animation", () => {
    render(<Card {...props} />);

    const container = screen.getByTestId("card-container");

    expect(container).toHaveStyle("opacity: 0");

    setTimeout(() => {
      expect(container).toHaveStyle("opacity: 1");
    }, 1000);
  });

  test("matches snapshot", () => {
    const { container } = render(<Card {...props} />);
    expect(container).toMatchSnapshot();
  });
});
