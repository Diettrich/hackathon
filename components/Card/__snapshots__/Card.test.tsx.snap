// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<div>
  <div
    class="container"
    data-testid="card-container"
    style="opacity: 0;"
  >
    <div
      class="pictureContainer"
    >
      <img
        alt="John Doe"
        class="picture"
        data-nimg="1"
        data-testid="card-picture"
        decoding="async"
        height="222"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
    <div
      class="nameContainer"
    >
      <h2
        class="name"
        data-testid="card-name"
      >
        John Doe
      </h2>
    </div>
    <div
      class="detailsContainer"
    >
      <div
        class="goal"
        data-testid="card-goal"
      >
        <span
          class="label"
        >
          Goal: 
        </span>
        <span
          class="value"
        >
          5000 $
        </span>
      </div>
      <div
        class="salary"
        data-testid="card-salary"
      >
        <span
          class="label"
        >
          Salary: 
        </span>
        <span
          class="value"
        >
          2500 $
        </span>
      </div>
    </div>
  </div>
</div>
`;
